### PR TITLE
Use baseUrl from config in frontend

### PR DESCRIPTION
**Pull request by BigBlue79**
## What changes did you make?

- added a schema to use `opslevel.baseUrl` in the frontend plugin. This is a required step, otherwise the value is inaccessible to the frontend plugin
- used the baseUrl in the few places we were hard-coding the OpsLevel url - the "View Full Maturity Report in OpsLevel" button on the `ExportEntitiesForm` at `/opslevel-maturity`

## Changelog

- [x] I have added a Changie entry or given a reason why this does not need an entry in this section.

## Screenshots

Screenshot 2024-01-24 at 6 32 21 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +51,14 @@ proxy:
 
 If you're running Self-Hosted OpsLevel, replace `target` with your URL.
 
+### Set Up the Base OpsLevel URL
```
Comment (patrick-th) [inline README.md:54]: just a drive-by: the backend plugin has the same fragment in its readme - we should probably update it there as well

**Comment by katerberg** [comment]
Still reviewing the code, but just dropping a line to point out that `src/test/CheckResultsByLevel.test.tsx` is failing.

File: src/components/ServiceMaturityReport.tsx [hunk 2]
```diff
@@ -37,15 +36,6 @@ export default function ServiceMaturityReport({
...
 }: Props) {
-  const { entity } = useEntity();
-  let opslevelUrl = "https://app.opslevel.com";
-  if (service) {
-    const extensionToRemove = `/services/${entity.metadata.name}`;
-    opslevelUrl = service.htmlUrl
-      .split(extensionToRemove)
-      .filter((s) => s)
-      .join(extensionToRemove);
-  }
```
Comment (katerberg) [inline src/components/ServiceMaturityReport.tsx:48]: So happy this is going away.

**Review by katerberg — COMMENTED** [review]
Code looks good, but going to wait to check it locally until the tests are sorted.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+    - [Submitting a change](#submitting-a-change)
+  - [Getting the code](#getting-the-code)
+    - [Installing git](#installing-git)
+    - [External contributors](#external-contributors)
+    - [OpsLevel contributors](#opslevel-contributors)
+  - [Local Development](#local-development)
+    - [Installation](#installation)
+    - [Changie (Change log generation)](#changie-change-log-generation)
+  - [Submitting a Pull Request](#submitting-a-pull-request)
+  - [Releasing](#releasing)
```
[the remark this answers is not in the document]
Comment (BigBlue79) [reply CONTRIBUTING.md:16]: I guess! I didn't touch this at all

File: README.md [hunk 4]
```diff
@@ -47,6 +51,14 @@ proxy:
...
+
+```yaml
+opslevel:
+  baseUrl: 'https://app.opslevel.com'
+```
+
+If you're running Self-Hosted OpsLevel, replace `target` with your URL.
```
[the remark this answers is not in the document]
Comment (BigBlue79) [reply README.md:61]: I think I'll keep it, but change it to reflect that you should change `baseUrl`

File: README.md [hunk 5]
```diff
@@ -47,6 +51,14 @@ proxy:
 
 If you're running Self-Hosted OpsLevel, replace `target` with your URL.
 
+### Set Up the Base OpsLevel URL
+
+```yaml
+opslevel:
+  baseUrl: 'https://app.opslevel.com'
```
Comment (katerberg) [inline README.md:58]: Do we have this as a PR on our staging deploy already or is that coming in a follow-up when we bump the version?

**Review by katerberg — APPROVED** [review]
Tested it and it works great!

File: README.md (hunk 5 above)
Comment (BigBlue79) [reply README.md:58]: We do now! Since this is a public repo, I'll share it elsewhere.
